src/main/webapp/dev/js/common/feedbackResponseComments.es6

```diff
@@ -457,8 +457,102 @@ function registerResponseCommentCheckboxEvent() {
+};
+
+function registerResponseCommentCheckboxEventForFeedbackPage() {
+    $('body').on('click', 'ul[id^="responseCommentTable"] * input[type=checkbox]', (e) => {
+        const table = $(e.currentTarget).closest('table');
+        const parentDiv = table.parent().attr('id');
+        const responseCommentId = parentDiv.substring('visibility-options-'.length);
+        let visibilityOptions = [];
+        const target = $(e.target);
+        const visibilityOptionsRow = target.closest('tr');
+        if (target.prop('class').includes('answerCheckbox') && !target.prop('checked')) {
+            visibilityOptionsRow.find('input[class*=giverCheckbox]').prop('checked', false);
+            visibilityOptionsRow.find('input[class*=recipientCheckbox]').prop('checked', false);
+        }
+        if ((target.prop('class').includes('giverCheckbox') || target.prop('class').includes('recipientCheckbox'))
+                && target.prop('checked')) {
+            visibilityOptionsRow.find('input[class*=answerCheckbox]').prop('checked', true);
+        }
+
+        table.find('.answerCheckbox:checked').each(function () {
+            visibilityOptions.push($(this).val());
+        });
+        $(`input[name=showresponsecommentsto-${responseCommentId}]`).val(visibilityOptions.join(', '));
+
+        visibilityOptions = [];
+        table.find('.giverCheckbox:checked').each(function () {
+            visibilityOptions.push($(this).val());
+        });
+        $(`input[name=showresponsegiverto-${responseCommentId}]`).val(visibilityOptions.join(', '));
+    });
+}
+
+function registerResponseCommentsEventForFeedbackPage() {
+    $('body').on('click',
+            'div[class*="responseCommentDeleteForm"] > a[id^="commentdelete"]', deleteCommentHandlerForFeedbackPage);
+    const clickHandlerMap = new Map();
+    clickHandlerMap.set(
+            '.show-frc-add-form', [showResponseCommentAddForm,
+                    ['recipientindex', 'giverindex', 'qnindex', 'sectionindex']]);
+    clickHandlerMap.set(
+            '.show-frc-edit-form', [showResponseCommentEditForm,
+                    ['recipientindex', 'giverindex', 'qnindex', 'frcindex', 'sectionindex', 'viewtype']]);
+    clickHandlerMap.set(
+            '.hide-frc-add-form', [hideResponseCommentAddForm,
+                    ['recipientindex', 'giverindex', 'qnindex', 'sectionindex']]);
+    clickHandlerMap.set(
+            '.hide-frc-edit-form', [hideResponseCommentEditForm,
+                    ['recipientindex', 'giverindex', 'qnindex', 'frcindex', 'sectionindex', 'viewtype']]);
+    clickHandlerMap.set(
+            '.toggle-visib-add-form', [toggleVisibilityAddForm,
+                    ['recipientindex', 'giverindex', 'qnindex', 'sectionindex']]);
+    clickHandlerMap.set(
+            '.toggle-visib-edit-form', [toggleVisibilityEditForm,
+                    ['recipientindex', 'giverindex', 'qnindex', 'frcindex', 'sectionindex', 'viewtype']]);
+
+    /* eslint-disable no-restricted-syntax */
```

Comment:
@chowyb This was added as `FeedbackSubmission` Pages doesn't have form unlike `InstructorResultsPage`.